src/classes/Addresses.cls

```diff
@@ -65,8 +65,33 @@ public inherited sharing class Addresses extends fflib_SObjects {
     }
 
     public void onAfterInsert() {
+        Map<Id, Address__c> defaultAndSeasonalAddressesByHouseholdAccountId =
```

Comment:
There's really nothing in the query of addresses that defines the result as "Default and Seasonal" - instead it's just all addresses. I recommend a clearer var name of `addressesByAccountId`.

Reply:
Line 69 is a call to a query method that filters the addresses in memory.  The query on Line 70 then gets the db versions of that subset.